### PR TITLE
Implement Style Library

### LINKED ISSUE #188: Implement RAC Style Library
Where possible, replace local styles with the [RAC Style Library](https://github.com/RockefellerArchiveCenter/styles) styles: [https://assets.rockarch.org/v0.9.1/main.min.css.](https://assets.rockarch.org/v0.8.1/main.min.css). 

Update UI components to follow the [RAC Style Guide
](https://styles.rockarch.org/?path=/docs/components-alert--docs)

### PR DESCRIPTION
**Pull request by p-galligan**
Fixes #188 . Implements style library.

### CODE REVIEW

File: index.html
Comment (HaSistrunk) [inline index.html]: You want to keep the library-specific text here.

File: _layouts/item.html
Comment (HaSistrunk) [inline _layouts/item.html]: To keep things maintainable and clear, instead of using inline styles in the HTML you'll want to define a CSS class here, and then use that class name in `custom.css` to define the styles.

File: index.html
Comment (p-galligan) [reply index.html]: Good catch! I missed that.

**Comment by HaSistrunk** [comment]
In terms of layout, it may be helpful to start out by adding a grid container div around the `<main>` element in `default.html` like this:
```
<div class="container grid">
    <main id="main">
        {{content}}
    </main>
</div>
```

And then adding grid-column styles in `custom` css for `#main`:

```
#main {
  grid-column: 2 / span 10;
}
```

That will set things off from the sides of the page pretty close to the original site layout.

**Comment by p-galligan** [comment]
Hannah and I had issues with getting the button correct on item detail pages. Would love another set of eyes on this.

**Comment by HaSistrunk** [comment]
This looks pretty good to me! I made a few updates to clean up unused styles, fix the layout issue with the button on the item details page, and update backstopjs selectors so that those tests continue to work (although if you've deleted data to make the build faster, you may have to add those back in for the backstop tests to work).

One thing I see is that we might want to make the modal a little smaller on the page, which can be accomplished by adding some local `.modal` styles. You can [see in the style library](https://github.com/RockefellerArchiveCenter/styles/blob/base/stylesheets/components/_modal.scss) that the size is governed by setting a `$modal-content-margin: 10%;`, and then using that value to set `top`, `bottom`, `left`, and `right`. values. So maybe we want to change that 10% to something higher. You can see this implemented in the [DIMES "confirm" modals](https://github.com/RockefellerArchiveCenter/dimes/blob/68e7a21d62156d17a810cec23aa3beec4f03c420/src/styles/components/_modal-confirm.scss).

**Comment by p-galligan** [comment]
> One thing I see is that we might want to make the modal a little smaller on the page, which can be accomplished by adding some local `.modal` styles.

I took a stab at this, ad it actually looks pretty good with the DIMES styles. Two things I noticed and am not sure how to fix:

1. Can we reduce the blank space on the bottom of the modal? It feels like a scroll shouldn't be necessary for the modal.
2. When you click on the calendar button on the modal it shifts all content on the page behind it (lol). Not sure what's happening there. I haven't noticed that before.

**Comment by p-galligan** [comment]
Screen Shot 2024-04-23 at 11 12 54 AM
Not ideal

**Comment by HaSistrunk** [comment]
> > One thing I see is that we might want to make the modal a little smaller on the page, which can be accomplished by adding some local `.modal` styles.
> 
> I took a stab at this, ad it actually looks pretty good with the DIMES styles. Two things I noticed and am not sure how to fix:
> 
> 1. Can we reduce the blank space on the bottom of the modal? It feels like a scroll shouldn't be necessary for the modal.
> 2. When you click on the calendar button on the modal it shifts all content on the page behind it (lol). Not sure what's happening there. I haven't noticed that before.

1. The DIMES styles set `bottom: auto;`. So you probably want to set the bottom attribute to also use the variable you've set for the other sides.]
2. Uhhh. Would have to troubleshoot that a bit - I'm not sure why that's changed from the original styles.

**Comment by HaSistrunk** [comment]
> 1. It feels like a scroll shouldn't be necessary for the modal.

You can get rid of that scrollbar by adjusting the [overflow attribute](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow) on the modal. So `overflow: hidden`.

**Review by HaSistrunk — APPROVED** [review]
This is looking good to me! In reviewing the backstop tests, I think most of the big changes I see are actually improvements. 

One thing we should look at holistically across sites is when we need to set a max-width on content for very large screens, but I don't think we need to figure that out now to address in this PR. I'l flag that to do some research on best practices.

When this is approved, let's update the backstop reference files before merging.

**Review by helrond — APPROVED** [review]
Love a diff that removes substantially more than it adds!

**Pull request merged**
